core/lib/da_client/src/lib.rs

```diff
@@ -0,0 +1,32 @@
+pub mod types;
+
+use std::fmt;
+
+use async_trait::async_trait;
+use types::{DAError, DispatchResponse, InclusionData};
+
+/// Trait that defines the interface for the data availability layer clients.
+#[async_trait]
+pub trait DataAvailabilityClient: Sync + Send + fmt::Debug {
+    /// Dispatches a blob to the data availability layer.
+    async fn dispatch_blob(
+        &self,
+        batch_number: u32,
+        data: Vec<u8>,
+    ) -> Result<DispatchResponse, DAError>;
```

Comment:
I assumed that no one would use more than 1 blob of DA layers now because their blobs are at least 2MB
but it can be extended in the future when there are multiple DA blobs per 1 batch